go/pvl/interp.go

```diff
@@ -129,73 +126,58 @@ func checkProofInner(g proofContextExt, pvlS string, service keybase1.ProofType,
 			"Bad signature: %v", err)
 	}
 
-	vars := scriptVariables{
-		UsernameService: info.RemoteUsername, // Empty "" for web and dns proofs
-		UsernameKeybase: info.Username,
-		Sig:             sigBody,
-		SigIDMedium:     sigID.ToMediumID(),
-		SigIDShort:      sigID.ToShortID(),
-		Hostname:        info.Hostname, // Empty "" except for web/dns proofs
-		Protocol:        info.Protocol, // Empty "" except for web proofs
+	scripts, perr := chunkGetScripts(&pvl, service)
+	if perr != nil {
+		return perr
 	}
 
-	// Enforce prooftype-dependent variables.
+	// validate hostname
 	webish := (service == keybase1.ProofType_DNS || service == keybase1.ProofType_GENERIC_WEB_SITE)
 	if webish {
-		vars.UsernameService = ""
-	} else {
-		vars.Hostname = ""
-	}
-	if service != keybase1.ProofType_GENERIC_WEB_SITE {
-		vars.Protocol = ""
-	}
-
-	// Validate and rewrite domain and protocol
-	if webish {
-		if !validateDomain(vars.Hostname) {
+		if !validateDomain(info.Hostname) {
 			return libkb.NewProofError(keybase1.ProofStatus_BAD_SIGNATURE,
-				"Bad hostname in sig: %s", vars.Hostname)
+				"Bad hostname in sig: %s", info.Hostname)
 		}
 	}
+
+	// validate protocol
 	if service == keybase1.ProofType_GENERIC_WEB_SITE {
-		canonicalProtocol, ok := validateProtocol(vars.Protocol, []string{"http", "https"})
-		if ok {
-			vars.Protocol = canonicalProtocol
-		} else {
+		_, ok := validateProtocol(info.Protocol, []string{"http", "https"})
+		if !ok {
 			return libkb.NewProofError(keybase1.ProofStatus_BAD_SIGNATURE,
-				"Bad protocol in sig: %s", vars.Protocol)
+				"Bad protocol in sig: %s", info.Protocol)
 		}
 	}
 
-	scripts, perr := chunkGetScripts(&pvl, service)
-	if perr != nil {
-		return perr
-	}
-
-	newstate := func(i int) scriptState {
+	mknewstate := func(i int) (scriptState, libkb.ProofError) {
```

Comment:
Tried it and I don't think it's simpler. Because it encapsulates ProofInfo, sigBody, sigID, and service. It ends up with either lots of params or two calls to `OpenSig`.